database/migrations/2022_05_11_201130_add_full_name_col.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+use Illuminate\Database\Migrations\Migration;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\Schema;
+
+class AddFullNameCol extends Migration
+{
+	private const USERS = 'users';
+	private const NAME = 'fullname';
+
+	/**
+	 * Run the migrations.
+	 *
+	 * @return void
+	 */
+	public function up()
+	{
+		Schema::table(self::USERS, function (Blueprint $table) {
+			$table->string(self::NAME, 128)->default('')->after('password')->nullable();
+		});
+	}
+
+	/**
+	 * Reverse the migrations.
+	 *
+	 * @return void
+	 */
+	public function down()
+	{
+		Schema::table(self::USERS, function (Blueprint $table) {
+			$table->dropColumn(self::NAME);
```

Comment:
Don't drop the column on downward migration. This breaks the foreign key constraints of the DB. This is due to a mix of a bug in Laravel and older versions of SQLite.

Reply:
I saw it but since dropping the column is in the migration of the email column as well, I thought it is the proper way to go.

So what should I put in instead?

Reply:
Although it is not great, the only feasible option is to do nothing on the downward path and simply keep the column. In exchange, you have to add a test in the upward path whether the column already exists.

Reply:
What you suggest is no solution at all, because it breaks the very concept of "migration".
What you suggest is simply "suppress the down migration" because your tools cannot handle them.

I am sorry, but this is nonsense. I tested the migrations with sqlite and they work. So we need to fix the workflow instead of breaking the concept of migration.

Reply:
> I am sorry, but this is nonsense. I tested the migrations with sqlite and they work. So we need to fix the workflow instead of breaking the concept of migration.

Yes they work, however they are silently dropping Foreign Key constraints in the case of SQL.
We discovered it at our own cost.